packages/flutter/lib/src/material/drawer.dart

```diff
@@ -217,7 +234,20 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
   }
 
   void _move(DragUpdateDetails details) {
-    final double delta = details.primaryDelta / _width;
+    if (_controller.isDismissed) {
+      final ModalRoute<dynamic> parentRoute = ModalRoute.of(context);
+      final bool canPop = parentRoute?.canPop ?? false;
+      if (canPop) {
+        Navigator.of(context).pop();
+        return;
+      }
+    }
+    double delta = details.primaryDelta / _width;
+    switch (widget.type) {
+      case DrawerAlignment.end:
+        delta = -delta;
+        break;
+    }
```

Comment:
I'm surprised the analyzer doesn't complain about the start case not being handled.  In any case, the idiom we prefer is to handle the start case (just `break;` is fine).

Reply:
Fixed